Fix index.json encoding

helfper opened the pull request:
`index.json` is currently overly escaped, which makes strange characters appear in the search context preview.

This also removes hidden pages from the search.

McShelby commented:
Could you make the removal of hidden pages from the index configurable?

At my company we use the hidden flag as a way to write draft content that doesn't show up in the menus but can be accessed by reviewers thru search. By that they don't have to type in the URL but can easily click the search result. This would get lost if I understand your comment correctly.

helfper commented:
I've added the site param `searchableHiddenPages`. Let me know if you prefer a different name, or I can just remove that part from `index.json`. I'm interested in the proper encoding part, but I don't really use hidden pages. I just added it because it seemed strange to see hidden pages in the search results.

McShelby commented:
Because most of the other options are in the form <VERB><NOUN> could you make it `searchHiddenPages`?

helfper commented:
@McShelby I've made the change.
